plugins/imagebase/plugin.js

```diff
@@ -0,0 +1,149 @@
+		var defaultTemplate = new CKEDITOR.template(
+			'<figure class="{basicClass}">' +
+				'<img alt="" src="" />' +
+				'<figcaption>{captionPlaceholder}</figcaption>' +
+			'</figure>' );
+
+		function joinClasses( basicClass, additionalClasses ) {
+			var toApply = additionalClasses.slice();
+
+			toApply.unshift( basicClass );
+
+			return toApply.join( ',' );
+		}
+
+		/**
+		 * This is an abstract class that describes a definition of an image widget.
+		 * It is a return type of {@link CKEDITOR.plugins.imagebase#createWidget} method.
+		 *
+		 * Note that because the image widget is a type of a widget, this definition extends
+		 * {@link CKEDITOR.plugins.widget.definition}.
+		 * It adds several parts of image and implements the {@link CKEDITOR.plugins.widget.definition#upcast} callback.
+		 * This callback should not be overwritten.
+		 *
+		 * @abstract
+		 * @class CKEDITOR.plugins.imagebase.imageWidgetDefinition
+		 * @mixins CKEDITOR.plugins.widget.definition
+		 */
+		return {
+			/**
+			 * @inheritdoc
+			 */
+			pathName: editor.lang.imagebase.pathName,
+
+			/**
+			 * @inheritdoc
+			 */
+			template: defaultTemplate,
+
+			/**
+			 * @inheritdoc
+			 */
+			allowedContent: {
+				img: {
+					attributes: '!src,alt,width,height'
+				},
+				figure: {
+					classes: '!' + joinClasses( options.basicClass, options.additionalClasses )
+				},
+				figcaption: true
+			},
+
+			/**
+			 * @inheritdoc
+			 */
+			requiredContent: 'figure(!' + options.basicClass + ')',
+
+			/**
+			 * @inheritdoc
+			 */
+			editables: {
+				caption: {
+					selector: 'figcaption',
+					pathName: editor.lang.imagebase.pathNameCaption,
+					allowedContent: options.captionAllowedContent
+				}
+			},
+
+			/**
+			 * @inheritdoc
+			 */
+			parts: {
+				image: 'img',
+				caption: 'figcaption'
+			},
+
+			/**
+			 * Image widget definition overwrites the {@link CKEDITOR.plugins.widget.definition#upcast} property.
+			 * This should not be changed.
+			 *
+			 * @property {Function}
+			 */
+			upcast: function( element ) {
+				// http://dev.ckeditor.com/ticket/11110 Don't initialize on pasted fake objects.
+				if ( element.attributes[ 'data-cke-realelement' ] ) {
+					return;
+				}
+
+				if ( element.name === 'figure' && element.hasClass( options.basicClass ) ) {
+					return element;
+				}
+			},
+
+			/**
+			 * Invokes user defined widget initialization method.
+			 *
+			 * @property {Function}
+			 */
+			init: function() {
+				if ( options.init ) {
+					options.init.call( this );
+				}
+			}
+		};
+	}
+
+	CKEDITOR.plugins.add( 'imagebase', {
+		requires: 'widget',
+		lang: 'en'
+	} );
+
+	/**
+	 * Namespace providing a set of helper functions for working with image widgets.
+	 *
+	 * @since 4.8.0
+	 * @singleton
+	 * @class CKEDITOR.plugins.imagebase
+	 */
+	CKEDITOR.plugins.imagebase = {
+		/**
+		 * Registers a new widget definition based on passed options.
+		 *
+		 * @since 4.8.0
```

Comment:
Since the parent class already h as a `since` tag I think it's redundant here, as it's the same version.